testkit/src/main/scala/org/bitcoins/testkit/dlc/DLCTest.scala

```diff
@@ -0,0 +1,773 @@
+        offerFundingInputs,
+        acceptFundingUtxos,
+        acceptFundingInputs,
+        feeRate,
+        timeouts
+      )
+    } else {
+      constructEnumDLCClients(
+        numOutcomesOrDigits,
+        oracleThreshold,
+        numOracles,
+        offerFundingPrivKey,
+        offerPayoutPrivKey,
+        acceptFundingPrivKey,
+        acceptPayoutPrivKey,
+        offerFundingUtxos,
+        offerFundingInputs,
+        acceptFundingUtxos,
+        acceptFundingInputs,
+        feeRate,
+        timeouts
+      )
+    }
+  }
+
+  def noEmptySPKOutputs(tx: Transaction): Boolean = {
+    tx.outputs.forall(_.scriptPubKey != EmptyScriptPubKey)
+  }
+
+  def setupDLC(
+      dlcOffer: TestDLCClient,
+      dlcAccept: TestDLCClient,
+      fundingTxF: Future[SetupDLC] => Future[Transaction],
+      publishTransaction: Transaction => Future[_])(implicit
+      ec: ExecutionContext): Future[(SetupDLC, SetupDLC)] = {
+    val offerSigReceiveP =
+      Promise[CETSignatures]()
+    val sendAcceptSigs = { sigs: CETSignatures =>
+      val _ = offerSigReceiveP.success(sigs)
+      FutureUtil.unit
+    }
+
+    val acceptSigReceiveP =
+      Promise[(CETSignatures, FundingSignatures)]()
+    val sendOfferSigs = {
+      (cetSigs: CETSignatures, fundingSigs: FundingSignatures) =>
+        val _ = acceptSigReceiveP.success(cetSigs, fundingSigs)
+        FutureUtil.unit
+    }
+
+    val acceptSetupF = dlcAccept.setupDLCAccept(sendSigs = sendAcceptSigs,
+                                                getSigs =
+                                                  acceptSigReceiveP.future)
+
+    val offerSetupF = dlcOffer.setupDLCOffer(getSigs = offerSigReceiveP.future,
+                                             sendSigs = sendOfferSigs,
+                                             getFundingTx =
+                                               fundingTxF(acceptSetupF))
+
+    for {
+      acceptSetup <- acceptSetupF
+      _ <- publishTransaction(acceptSetup.fundingTx)
+      offerSetup <- offerSetupF
+    } yield {
+      assert(acceptSetup.fundingTx == offerSetup.fundingTx)
+      assert(acceptSetup.refundTx == offerSetup.refundTx)
+      acceptSetup.cets.foreach {
+        case (msg, cetInfo) =>
+          assert(cetInfo.tx == offerSetup.cets(msg).tx)
+      }
+
+      (offerSetup, acceptSetup)
+    }
+  }
+
+  /** Computes an EnumOracleSignature for the given outcome and oracle */
+  def genEnumOracleSignature(
+      oracleInfo: EnumSingleOracleInfo,
+      outcome: String,
+      privKey: ECPrivateKey = oraclePrivKey,
+      kVal: ECPrivateKey = preCommittedK): EnumOracleSignature = {
+    val sig = privKey.schnorrSignWithNonce(CryptoUtil
+                                             .sha256DLCAttestation(outcome)
+                                             .bytes,
+                                           kVal)
+
+    EnumOracleSignature(oracleInfo, sig)
+  }
+
+  def genEnumOracleOutcome(
+      chosenOracles: Vector[Int],
```

Comment:
With over-sharing of oracles added, should make it sometimes use a larger set of oracles

Reply:
added a todo